src/analysis/convex/segment.lean

```diff
@@ -306,6 +306,29 @@ begin
   rw [←sub_eq_neg_add, ←neg_sub, hxy, ←sub_eq_neg_add, hzx, smul_neg, smul_comm, neg_add_self]
 end
 
+open affine_map
+
+lemma open_segment_subset_union (c : 𝕜) (x y : E) :
+  open_segment 𝕜 x y ⊆ insert (line_map x y c) (open_segment 𝕜 x (line_map x y c) ∪
+    open_segment 𝕜 (line_map x y c) y) :=
```

Comment:
Wouldn't this be better to state this for any element aligned with `x` and `y` rather than syntactically `line_map x y c`? It would be annoying if you had to turn a random point into an image of the line map before being able to use this lemma.

Reply:
What predicate do you mean by "any element aligned with `x` and `y`"?

Reply:
Yeah it doesn't exist, does it? 😓 and it would be misbehaved when `x = y`.

Reply:
I reformulated using `set.range`.